src/security/srv_acl.c

```diff
@@ -506,16 +506,65 @@ get_sec_capas_for_token(Auth__Token *token, struct ownership *ownership,
 	return rc;
 }
 
+static int
+get_sec_origin_for_token(Auth__Token *token, char **machine)
+{
+	struct drpc_alloc	alloc = PROTO_ALLOCATOR_INIT(alloc);
+	int 			machine_size = 0;
+	int 			rc = 0;
+	char 			*mtmp;
+	Auth__Sys		*authsys;
+
+
+	if (token == NULL || machine == NULL) {
+		D_ERROR("NULL input\n");
+		return -DER_INVAL;
+	}
+
+	rc = get_auth_sys_payload(token, &authsys);
+	if (rc != 0)
+		return rc;
+
+	if(authsys->machinename) {
+		 machine_size = strnlen(authsys->machinename, MAXHOSTNAMELEN);
+	} else {
+		D_ERROR("Malformed AuthSys token missing machinename");
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	if (machine_size > 64) {
+		D_ERROR("hostname provided by the agent is too large");
+		rc = -DER_INVAL;
+		goto out;
+	}
+
+	D_STRNDUP(mtmp, authsys->machinename, machine_size);
+
+	if (mtmp) {
+		*machine = mtmp;
+		rc = 0;
+	} else {
+		rc = -DER_NOMEM;
+	}
+
+out:
+	auth__sys__free_unpacked(authsys, &alloc.alloc);
+	return rc;
+}
+
 int
 ds_sec_pool_get_capabilities(uint64_t flags, d_iov_t *cred,
 			     struct ownership *ownership,
-			     struct daos_acl *acl, uint64_t *capas)
+			     struct daos_acl *acl, uint64_t *capas,
+			     char **machine)
```

Comment:
This addition feels weird to me, it's totally outside of the purpose of digesting the capabilities based on the ACL and flags. I'd be in favor of a new function to extract the machine name, even if it has to unpack the token again. This is only done at pool connect so not on the hot IO path.